src/core/core.js

```diff
@@ -540,4 +542,73 @@ p5.prototype.registerMethod = function(name, m) {
   p5.prototype._registeredMethods[name].push(m);
 };
 
+p5.prototype._createFriendlyGlobalFunctionBinder = function(options) {
+  options = options || {};
+
+  var globalObject = options.globalObject || window;
+  var log = options.log || console.log.bind(console);
+  var propsToForciblyOverwrite = {
+    // p5.print actually always overwrites an existing global function,
+    // albeit one that is very unlikely to be used:
+    //
+    //   https://developer.mozilla.org/en-US/docs/Web/API/Window/print
+    'print': true
+  };
+
+  return function(prop, value) {
+    if (typeof(value) === 'function') {
+      try {
+        // Because p5 has so many common function names, it's likely
+        // that users may accidentally overwrite global p5 functions with
+        // their own variables. Let's allow this but log a warning to
+        // help users who may be doing this unintentionally.
+        //
+        // For more information, see:
+        //
+        //   https://github.com/processing/p5.js/issues/1317
+
+        if (prop in globalObject && !(prop in propsToForciblyOverwrite)) {
+          throw new Error('global "' + prop + '" already exists');
```

Comment:
Oh, good question... I actually ran into this problem before adding the explicit `throw` in lines 570-571, when I used the following sketch:

```
var text;

function setup() {}
```

Basically browsers threw an actual error when I was trying to call `Object.defineProperty`, claiming that I was trying to "redefine a non-configurable property". So I originally added the try/catch to detect for that, but I ended up later adding lines 570-571, which check for such a condition for us, so we might be able to get rid of the try/catch...

That said, because this functionality will *always* be called in global mode, and because it's sort of "nice to have" functionality rather than "must have" functionality, I thought it might be a good idea to leave the try/catch with the fallback to legacy behavior in there just in case there are really weird edge cases that we don't know about... I can remove it if you want though!